packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -227,34 +190,34 @@ describeNoCompat("GC Random tests", (getTestObjectProvider) => {
 
     const getTestNode = (id: string): TestNode => {
         const testNode = testNodes.get(id);
-        assert(testNode !== undefined, `Datastore ${id} should exist in datastores!`);
+        assert(testNode !== undefined, `Datastore ${id} not found in TestNode map!`);
         return testNode;
     };
 
-    const updateReferencedDataStores = (id: string) => {
-        if (!refDataStores.has(id)) {
-            refDataStores.add(id);
-            unrefDataStores.delete(id);
-            for (const childId of getTestNode(id).children) {
-                updateReferencedDataStores(childId);
+    const trackDataStoreAsReferenced = (id: string) => {
+        const testNode = getTestNode(id);
+        if (testNode.referenceState === ReferenceState.Referenced) {
```

Comment:
Should be `ReferenceState.Unreferenced`?

Reply:
or `!==` I suppose

Reply:
Good catch